app/controllers/case_contacts/form_controller.rb

```diff
@@ -65,6 +64,17 @@ def get_cases_and_contact_types
     @selected_contact_type_ids = @case_contact.contact_type_ids
   end
 
+  def manage_navigation
+    if @nav_step.present?
+      jump_to(@nav_step.split("/").last.to_sym)
+      render_wizard @case_contact, {}, {case_contact_id: @case_contact.id}
```

Comment:
Used [wicked](https://github.com/zombocom/wicked) `jump_to` and then renders wizard to go to the specific step from navigation buttons